pkg/prometheus/collector.go

```diff
@@ -29,6 +29,14 @@ var (
 			"name",
 		}, nil,
 	)
+	descPrometheusEnforcedSampleLimit = prometheus.NewDesc(
+		"prometheus_operator_enforced_sample_limit",
+		"Global limit on the number of scraped samples per scrape target.",
+		[]string{
+			"namespace",
+			"name",
```

Comment:
I believe this is Prometheus name, wonder if this is descriptive enough of a label name? @nrchakradhar @simonpasquier wdyt?

Reply:
This is consistent with the labels of `prometheus_operator_spec_replicas` above. Maybe the metric name should be `prometheus_operator_spec_enforced_sample_limit` after all?

Reply:
I think `prometheus_operator_prometheus_enforced_sample_limit ` maybe?

Reply:
This name looks more reasonable. Changed it.